packages/typeorm-decorator-ismoney/src/index.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+} from 'class-validator';
+
+import codes from './codes';
+import { IMoney } from './Money';
+
+const settingsDefault = {
+  allowNegative: false,
+  min: Number.MIN_SAFE_INTEGER,
+  max: Number.MAX_SAFE_INTEGER,
+  currencies: null,
+};
+
+interface IMoneyValidationOptions {
+  allowNegative?: boolean;
+  min?: number;
+  max?: number;
+  currencies?: string[];
+}
+
+@ValidatorConstraint({ name: 'IsMoney', async: false })
+export class IsMoneyValidator implements ValidatorConstraintInterface {
+  public validate(money: IMoney, args: ValidationArguments) {
+    const { amount, currency } = money;
+    const settingsOrDefault = args.constraints[0] || settingsDefault;
+    const {
+      allowNegative,
+      min,
+      max,
+      currencies: customCurrencies,
+    } = Object.assign({}, settingsDefault, settingsOrDefault);
+    const isAmountCorrect =
+      typeof amount === 'number' &&
+      (allowNegative || amount >= 0) &&
+      amount <= max &&
+      amount >= min;
+    const isCurrencyCorrect =
+      typeof currency === 'string' && codes.includes(currency);
+    const isCustomCurrencyCorrect =
+      !customCurrencies ||
+      (customCurrencies.length && customCurrencies.includes(currency));
+    return isAmountCorrect && isCurrencyCorrect && isCustomCurrencyCorrect;
+  }
+
+  public defaultMessage(args: ValidationArguments) {
+    // here you can provide default error message if validation failed
+    return `Provided object (${JSON.stringify(
+      args.value,
+    )}) is too a valid Money object`;
```

Comment:
looks like something wrong with the message

Reply:
It is. Fixed wording.